### PR TITLE
util/codec: improve package test code coverage above 85%

### PR DESCRIPTION
**Pull request by tiancaiamao**
### What problem does this PR solve? 

After this commit:

```
ok  	github.com/pingcap/tidb/util/codec	0.054s	coverage: 85.4% of statements
```

### What is changed and how it works?

Add test to cover more code

### Check List 

Tests 

 - Unit test

### CODE REVIEW

File: util/codec/codec.go [hunk 1]
```diff
@@ -205,15 +206,11 @@ func encodeChunkRow(sc *stmtctx.StatementContext, b []byte, row chunk.Row, allTy
 				break
 			}
 			// encode unsigned integers.
-			if hash {
```
Comment (tiancaiamao) [inline util/codec/codec.go]: `hash` is always true in the codebase, so I remove the other branch to improve test coverage.

File: util/codec/codec.go [hunk 2]
```diff
@@ -246,20 +243,13 @@ func encodeChunkRow(sc *stmtctx.StatementContext, b []byte, row chunk.Row, allTy
 			b = EncodeInt(b, int64(row.GetDuration(i, 0).Duration))
 		case mysql.TypeNewDecimal:
 			b = append(b, decimalFlag)
-			if hash {
```
Comment (tiancaiamao) [inline util/codec/codec.go]: ditto

File: util/codec/codec.go [hunk 3]
```diff
@@ -510,12 +500,7 @@ func peekBytes(b []byte, reverse bool) (int, error) {
 		// The byte slice is encoded into many groups.
 		// For each group, there are 8 bytes for data and 1 byte for marker.
 		marker := b[offset+encGroupSize]
-		var padCount byte
-		if reverse {
```
Comment (tiancaiamao) [inline util/codec/codec.go]: `reverse` is always false, so I remove the other branch to improve test coverage

**Comment by tiancaiamao** [comment]
PTAL @winkyao @jackysp

**Comment by tiancaiamao** [comment]
PTAL @zz-jason

**Comment by jackysp** [comment]
PTAL @winkyao @zz-jason

**Pull request merged**